pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PublishRateLimiterTest.java

```diff
@@ -104,9 +104,13 @@ public void testPrecisePublishRateLimiterAcquire() throws Exception {
 
         // tryAcquire msgSizeInBytes exceeded
         assertFalse(precisPublishLimiter.tryAcquire(10, 101));
-        Thread.sleep(1100);
+        Thread.sleep(2100);
```

Comment:
This is a potential flaky test, and we cannot guarantee that it will be scheduled after 2100ms. Manual control of renew should be better

Reply:
I can change it, didn't want to touch the current implementation too much.

But will do 👍🏾

Reply:
Would you like me to turn the rateLimiters within the precisPublishLimiter to public in order to call the renew function?

Im afraid that it can result others to try and access the rateLimiters state, Oppositely to OOP mutable non-shared state approach

Reply:
We can use reflection.
Stop the scheduled task first
private ScheduledFuture<?> renewTask;

Reply:
Oh sounds good, will do thanks for clarifying

Reply:
Done 👍🏾